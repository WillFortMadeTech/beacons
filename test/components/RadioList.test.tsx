import { render } from "@testing-library/react";
import React from "react";
import {
  RadioList,
  RadioListItem,
  RadioListItemHint,
} from "../../src/components/RadioList";

describe("Radio List component", () => {
  it("renders correctly with div inside", () => {
    const { asFragment } = render(
      <RadioList>
        <div></div>
      </RadioList>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with radio list item inside", () => {
    const { asFragment } = render(
      <RadioList>
        <RadioListItem
          id="test001"
          name="test001"
          text="test001"
          value="test001"
        />
      </RadioList>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with hint text", () => {
    const { asFragment } = render(
      <RadioList>
        <RadioListItemHint
          id="test001"
          name="test001"
          text="test001"
          value="test001"
          hintText="test001"
        />
      </RadioList>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
