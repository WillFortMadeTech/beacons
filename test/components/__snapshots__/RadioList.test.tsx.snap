// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio List component renders correctly with div inside 1`] = `
<DocumentFragment>
  <div
    class="govuk-radios "
  >
    <div />
  </div>
</DocumentFragment>
`;

exports[`Radio List component renders correctly with radio list item inside 1`] = `
<DocumentFragment>
  <div
    class="govuk-radios "
  >
    <div
      class="govuk-radios__item"
    >
      <input
        aria-describedby="test001-hint"
        class="govuk-radios__input"
        id="test001"
        name="test001"
        type="radio"
        value="test001"
      />
      <label
        class="govuk-label govuk-radios__label"
        for="test001"
      >
        test001
      </label>
    </div>
  </div>
</DocumentFragment>
`;
